### PR TITLE
Handling 429 responses from Public API

### PR DESCRIPTION
**Pull request by mateusz-sekara**

### CODE REVIEW

File: src/utils/errors.ts [hunk 1]
```diff
@@ -36,16 +36,38 @@ const requestQuery = (response: HttpResponse): string => {
...
     statusCode: response.status,
-    statusMessage: extractStatusMessage(errorResponse),
-    errorDetails: extractErrorDetails(errorResponse),
+    ...buildErrorMessageDetails(z, response),
     query: requestQuery(response),
     url: response.request.url
   }
 }
 
+export const buildErrorMessageDetails = (z: ZObject, response: HttpResponse): { statusMessage: string, errorDetails: string[] } => {
```
Comment (kozielt) [inline src/utils/errors.ts]: Pretty long line, eslint does not complain about it?

File: src/utils/errors.ts [hunk 2]
```diff
@@ -36,16 +36,38 @@ const requestQuery = (response: HttpResponse): string => {
...
+    return {
+      statusMessage: '429 Too Many Requests',
+      errorDetails: ['You reached rate limit for Sell API']
+    }
+  }
+  // Purpose of this layer is to not break Zapier App in case of returning malformed envelope from Public API
+  try {
+    const errorResponse = parseErrorResponse(z, response.content)
+    return {
+      statusMessage: extractStatusMessage(errorResponse),
```
Comment (kozielt) [inline src/utils/errors.ts]: We're handling all statuses `extractStatusMessage` and `extractErrorDetails` but we need to add separate if for `isRateLimitReached`? I'm wondering if this should be handled in one place

File: src/utils/errors.ts (hunk 1 above)
Comment (mateusz-sekara) [reply src/utils/errors.ts]: Hmm, probably I have misconfigured it, will doubl check configuration

File: src/utils/errors.ts (hunk 2 above)
Comment (mateusz-sekara) [reply src/utils/errors.ts]: Yep, you are right. Will move all those to single place

**Pull request merged**